allennlp/nn/beam_search.py

```diff
@@ -111,7 +135,8 @@ def search(
         # beam to `beam_size`^2 candidates from which we will select the top
         # `beam_size` elements for the next iteration.
         # shape: (batch_size, num_classes)
-        start_class_log_probabilities, state = step(start_predictions, start_state, 0)
+        with torch.no_grad():
```

Comment:
If you never want any grads in here, you can annotate the whole function with `no_grad`.

Reply:
Neat, I didn't know this.